api: directly query for revisions when merging

slint opened the pull request:
* Instead of relying on the array access for fetching
  SQLAlchemy-Continuum revisions when doing a merge publish, we now
  directly query for the revision that was stored. This is done
  because there are cases where revision storing is skipped and thus
  array item index access doesn't match one-to-one with what is stored
  in the database.

ntarocco commented on invenio_deposit/api.py:
```diff
@@ -171,7 +172,13 @@ def fetch_published(self):
+        # NOTE: revision_id = version_id - 1
+        lca_version_id = self['_deposit']['pid']['revision_id'] + 1
+        lca = RecordRevision(
+            first.model.versions.filter(version_id=lca_revision_id).one())
```
```suggestion
            first.model.versions.filter_by(version_id=lca_version_id).one())
```

slint committed:
api: directly query for revisions when merging

* Instead of relying on the array access for fetching
  SQLAlchemy-Continuum revisions when doing a merge publish, we now
  directly query for the revision that was stored. This is done
  because there are cases where revision storing is skipped and thus
  array item index access doesn't match one-to-one with what is stored
  in the database.

ntarocco reviewed (approved):
How do we do with all the build failures now? :)

slint commented:
This PR is obsolete now since this is addressed properly in the base Record API class in `invenio-records v1.3.2` by https://github.com/inveniosoftware/invenio-records/pull/221
